src/screens/CreateAuthenticator/CreateAuthenticatorScreen.tsx

```diff
@@ -56,18 +81,24 @@ class CreateAuthenticatorScreen extends Component<Props> {
     ]);
   };
 
-  createAuthenticator = (json: string) => {
+  createAuthenticator = () => {
     const { navigation, createAuthenticator } = this.props;
     try {
-      const data = JSON.parse(json);
-      if (!data.entropy || !data.name) {
+      if (!this.state.label) {
         throw new Error('Invalid data');
       }
-      createAuthenticator(data, {
-        onSuccess: (authenticator: IAuthenticator) => navigation.navigate(Route.EnterPIN, { id: authenticator.id }),
-        onFailure: (error: string) => this.onCreateAuthenticatorFailure(error),
-      });
+      createAuthenticator(
+        { name: this.state.label, mnemonic: '' },
+        {
+          onSuccess: (authenticator: IAuthenticator) => {
+            navigation.goBack();
+            navigation.navigate(Route.CreateAuthenticatorPublicKey, { id: authenticator.id });
+          },
+          onFailure: (error: string) => this.onCreateAuthenticatorFailure(error),
+        },
+      );
     } catch (_) {
+      // TODO: change text for error
       this.onCreateAuthenticatorFailure(i18n.wallets.errors.invalidPrivateKey);
```

Comment:
you can delete this without the condition that throws new Error('Invalid data'); (see comment above)